conda_build/post.py

```diff
@@ -45,14 +46,22 @@ def fix_shebang(f, osx_is_app=False):
         return
     elif os.path.islink(path):
         return
-    with io.open(path, encoding=locale.getpreferredencoding()) as fi:
+
+    with io.open(path, encoding=locale.getpreferredencoding(), mode='r+') as fi:
         try:
-            data = fi.read()
-        except UnicodeDecodeError: # file is binary
+            data = fi.read(100)
```

Comment:
yes, I did a few checks and with the file open invocation, even fh.read(1) gives something like

```
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf3 in position 1: ordinal not in range(128)
```

But, I'm not positive how fragile this is without reading the entire file.